tools/internal_ci/linux/pull_request/grpc_tsan_on_foundry.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+EXTRA_FLAGS="--copt=-gmlt --strip=never --copt=-fsanitize=thread --linkopt=-fsanitize=thread --test_timeout=3600 --action_env=TSAN_OPTIONS=suppressions=test/core/util/tsan_suppressions.txt:halt_on_error=1:second_deadlock_stack=1"
```

Comment:
3600 seconds is one hour. Do we really need such a massive timeout?

Reply:
Most likely not because the whole test ends within 15min or so. But it's the timeout used for all sanitizers. If we want to use a smaller one, I need to check all sanitizer tests, choose an appropriate one and make sure there's no timeouts, which is out of the scope of this PR. Plus I don't this hurts us because if the tests take shorter, this flag will be a no-op.